ethereum/core/src/main/java/tech/pegasys/pantheon/ethereum/chain/DefaultMutableBlockchain.java

```diff
@@ -192,18 +229,25 @@ private UInt256 calculateTotalDifficulty(final Block block) {
   private BlockAddedEvent updateCanonicalChainData(
       final BlockchainStorage.Updater updater,
       final Block newBlock,
-      final UInt256 totalDifficulty) {
+      final UInt256 totalDifficulty,
+      final boolean isFastSync) {
     final Hash chainHead = blockchainStorage.getChainHead().orElse(null);
     if (newBlock.getHeader().getNumber() != BlockHeader.GENESIS_BLOCK_NUMBER && chainHead == null) {
       throw new IllegalStateException("Blockchain is missing chain head.");
     }
```

Comment:
I think you need a check somewhere to make sure we're not interleaving regular blocks with fast-sync blocks, which could leave the chain data in an inconsistent state.

Reply:
I think the code around this prevents that anyway but I've added an explicit check to help us sleep better at night.